src/ORM/DataQuery.php

```diff
@@ -394,6 +394,14 @@ protected function ensureSelectContainsOrderbyColumns($query, $originalSelect =
                     continue;
                 }
 
+                // Leave it alone if we're already selecting it with an expression,
+                // for example using a case statement, which may select values from various tables.
+                // This regex looks for anything that isn't a single column, with optional table name and ANSI quotes.
+                $selects = $query->getSelect();
+                if (isset($selects[$col]) && !preg_match('/^\s*"?[^."]+"?\.?"?[^"]*"?\s*$/', $selects[$col])) {
```

Comment:
```suggestion
                if (isset($selects[$col]) && !preg_match('/^\s*"?[^\(."]+"?\.?"?[^\("]*"?\s*$/', $selects[$col])) {
```
Check for function calls as well.